### ISSUE TITLE
clusterctl describe cluster  throws nil pointer exception is cluster is still provisioning

### ISSUE THREAD

**Issue by arianvp** [kubernetes-sigs/cluster-api#6333]
**What steps did you take and what happened:**
1. create a cluster
2. quickly run `clusterctl describe cluster`
```
$ clusterctl describe cluster sample
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1436dd9]

goroutine 1 [running]:
sigs.k8s.io/cluster-api/util.UnstructuredUnmarshalField(0x0, {0x1992d20, 0xc0008720d8}, {0xc0009174e8, 0x2, 0x2})
	sigs.k8s.io/cluster-api/util/util.go:407 +0x59
sigs.k8s.io/cluster-api/util/conditions.(*unstructuredWrapper).GetConditions(0xc000114a88)
	sigs.k8s.io/cluster-api/util/conditions/unstructured.go:55 +0x99
sigs.k8s.io/cluster-api/util/conditions.Get({0x1ef77b8, 0xc000114a88}, {0x1bcf97e, 0x5})
	sigs.k8s.io/cluster-api/util/conditions/getter.go:39 +0x35
sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree.GetReadyCondition({0x1ef6720, 0x0})
	sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree/util.go:36 +0xb6
sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree.ObjectTree.Add({{0x1ef6848, 0xc0001d48c0}, {{0x0, 0x0}, 0x0, 0x0, 0x1}, 0xc000711230, 0xc000711380}, {0x1ef6720, ...}, ...)
	sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree/tree.go:85 +0xfa
sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree.Discovery.func1({0x1ef6720, 0x0}, 0xc0007d1b00)
	sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree/discovery.go:91 +0x191
sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree.Discovery({0x1e95af0, 0xc00005a218}, {0x1ecec78, 0xc000438850}, {0x1bd308b, 0x7}, {0x7ffef0d2f68e, 0x6}, {{0x0, 0x0}, ...})
	sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree/discovery.go:108 +0x639
sigs.k8s.io/cluster-api/cmd/clusterctl/client.(*clusterctlClient).DescribeCluster(0xc0000ce9c0, {{{0x0, 0x0}, {0x0, 0x0}}, {0x1bd308b, 0x7}, {0x7ffef0d2f68e, 0x6}, {0x0, ...}, ...})
	sigs.k8s.io/cluster-api/cmd/clusterctl/client/describe.go:82 +0x1c5
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.runDescribeCluster({0x7ffef0d2f68e, 0x6})
	sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/describe_cluster.go:144 +0x198
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.glob..func5(0x2e82220, {0xc00048a610, 0x1, 0x1})
	sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/describe_cluster.go:96 +0x2a
github.com/spf13/cobra.(*Command).execute(0x2e82220, {0xc00048a5f0, 0x1, 0x1})
	github.com/spf13/cobra@v1.2.1/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x2e847a0)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.Execute()
	sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/root.go:94 +0x25
main.main()
	sigs.k8s.io/cluster-api/cmd/clusterctl/main.go:26 +0x17

```

**What did you expect to happen:**

No nil pointer exception

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version: v1.1.3
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`): `Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.4", GitCommit:"e6c093d87ea4cbb530a7b2ae91e54c0842d8308a", GitTreeState:"archive", BuildDate:"1980-01-01T00:00:00Z", GoVersion:"go1.17.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:52:18Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
`
- OS (e.g. from `/etc/os-release`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by killianmuldoon** [comment]
I think #6334 should be enough to fix this - but if you could show us the objects that would be great to understand how the code got here.

**Comment by arianvp** [comment]
Cluster:
```
apiVersion: cluster.x-k8s.io/v1beta1
kind: Cluster
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"cluster.x-k8s.io/v1beta1","kind":"Cluster","metadata":{"annotations":{},"name":"sample","namespace":"default"},"spec":{"controlPlaneRef":{"apiVersion":"controlplane.cluster.x-k8s.io/v1beta1","kind":"KubeadmControlPlane","name":"sample"},"infrastructureRef":{"apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha1","kind":"VultrCluster","name":"sample"}}}
  creationTimestamp: "2022-03-24T15:10:14Z"
  finalizers:
  - cluster.cluster.x-k8s.io
  generation: 3
  name: sample
  namespace: default
  resourceVersion: "287894"
  uid: 2039a36b-0749-4b60-8005-34665c8831f9
spec:
  clusterNetwork:
    pods:
      cidrBlocks:
      - 2001:19f0:6c01:106c:0000:0000:0000:0000/64
    services:
      cidrBlocks:
      - fdc8:fa6b:6163:dcc8::/108
  controlPlaneEndpoint:
    host: 2001:19f0:6c01:08fd:ffff:ffff:ffff:ffff
    port: 6443
  controlPlaneRef:
    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
    kind: KubeadmControlPlane
    name: sample
    namespace: default
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
    kind: VultrCluster
    name: sample
    namespace: default
status:
  conditions:
  - lastTransitionTime: "2022-03-24T15:16:09Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2022-03-24T15:16:09Z"
    status: "True"
    type: InfrastructureReady
  infrastructureReady: true
  observedGeneration: 3
  phase: Provisioned
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
kind: VultrCluster
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha1","kind":"VultrCluster","metadata":{"annotations":{},"name":"sample","namespace":"default"},"spec":{"loadBalancer":{},"podsReservedIP":{},"region":"fra","startupScript":{}}}
  creationTimestamp: "2022-03-24T15:10:14Z"
  finalizers:
  - cluster.cluster.x-k8s.io
  generation: 3
  labels:
    cluster.x-k8s.io/cluster-name: sample
  name: sample
  namespace: default
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Cluster
    name: sample
    uid: 2039a36b-0749-4b60-8005-34665c8831f9
  resourceVersion: "287889"
  uid: d57605ed-03eb-4098-a52b-d7621890cb1b
spec:
  controlPlaneEndpoint:
    host: 2001:19f0:6c01:08fd:ffff:ffff:ffff:ffff
    port: 6443
  loadBalancer:
    resourceId: 1c73ad23-39d9-4012-b6c9-9b685d179da5
  podsReservedIP:
    resourceId: c2a1d160-559b-4301-812d-63fbef1a58d1
  region: fra
  startupScript:
    resourceId: ac921922-fa07-4e29-ab2e-bec5db55ffa9
status:
  loadBalancer:
    address: 2001:19f0:6c01:08fd:ffff:ffff:ffff:ffff
    port: 6443
    status: active
  podsCIDR: 2001:19f0:6c01:106c:0000:0000:0000:0000/64
  ready: true
```


Machine:
```
apiVersion: cluster.x-k8s.io/v1beta1
kind: Machine
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"cluster.x-k8s.io/v1beta1","kind":"Machine","metadata":{"annotations":{},"labels":{"cluster.x-k8s.io/cluster-name":"sample","cluster.x-k8s.io/control-plane":"true","set":"controlplane"},"name":"sample","namespace":"default"},"spec":{"bootstrap":{"configRef":{"apiVersion":"bootstrap.cluster.x-k8s.io/v1beta1","kind":"KubeadmConfig","name":"sample"}},"clusterName":"sample","infrastructureRef":{"apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha1","kind":"VultrMachine","name":"sample"},"version":"v1.23.5"}}
  creationTimestamp: "2022-03-24T15:56:11Z"
  finalizers:
  - machine.cluster.x-k8s.io
  generation: 3
  labels:
    cluster.x-k8s.io/cluster-name: sample
    cluster.x-k8s.io/control-plane: "true"
    set: controlplane
  name: sample
  namespace: default
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    kind: Cluster
    name: sample
    uid: 2039a36b-0749-4b60-8005-34665c8831f9
  resourceVersion: "299219"
  uid: bdf6feb1-9fde-4590-b5e7-9f952f0a7cb0
spec:
  bootstrap:
    configRef:
      apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
      kind: KubeadmConfig
      name: sample
      namespace: default
    dataSecretName: sample
  clusterName: sample
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
    kind: VultrMachine
    name: sample
    namespace: default
  providerID: e4fd9297-3c60-4e78-bf6f-24a0c364b207
  version: v1.23.5
status:
  addresses:
  - address: 2001:19f0:6c01:14ca:5400:03ff:feeb:770d
    type: ExternalIP
  bootstrapReady: true
  conditions:
  - lastTransitionTime: "2022-03-24T15:56:44Z"
    status: "True"
    type: Ready
  - lastTransitionTime: "2022-03-24T15:56:12Z"
    status: "True"
    type: BootstrapReady
  - lastTransitionTime: "2022-03-24T15:56:44Z"
    status: "True"
    type: InfrastructureReady
  - lastTransitionTime: "2022-03-24T15:56:11Z"
    reason: WaitingForNodeRef
    severity: Info
    status: "False"
    type: NodeHealthy
  infrastructureReady: true
  lastUpdated: "2022-03-24T15:56:44Z"
  observedGeneration: 2
  phase: Provisioned
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
kind: VultrMachine
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"infrastructure.cluster.x-k8s.io/v1alpha1","kind":"VultrMachine","metadata":{"annotations":{},"name":"sample","namespace":"default"},"spec":{"plan":"vc2-2c-4gb","region":"fra"}}
  creationTimestamp: "2022-03-24T15:56:11Z"
  finalizers:
  - cluster.cluster.x-k8s.io
  generation: 2
  labels:
    cluster.x-k8s.io/cluster-name: sample
  name: sample
  namespace: default
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Machine
    name: sample
    uid: bdf6feb1-9fde-4590-b5e7-9f952f0a7cb0
  resourceVersion: "299216"
  uid: 4bcb1dad-4757-4c49-b3cb-e8e50a321e72
spec:
  plan: vc2-2c-4gb
  providerID: e4fd9297-3c60-4e78-bf6f-24a0c364b207
  region: fra
status:
  addresses:
  - address: 2001:19f0:6c01:14ca:5400:03ff:feeb:770d
    type: ExternalIP
  ready: true
  status: active
```

**Comment by arianvp** [comment]
Could somebody tell me what's missing in my above definition that created the `nil` exception?

**Comment by killianmuldoon** [comment]
I haven't had time to fully recreate this issue - but I think the source is the lack of conditions on your VultrMachine object.
`sigs.k8s.io/cluster-api/cmd/clusterctl/client/tree.GetReadyCondition` is looking for a ready condition on the machine, and the nil pointer is happening as a result of not finding it.

**Comment by killianmuldoon** [comment]
@arianvp Did you  manage to solve this issue?

**Comment by killianmuldoon** [comment]
I think we can close this given a lack of updates and assuming that it was fixed, as expected, by #6334

/close

**Comment by killianmuldoon** [comment]
I think we can close this given a lack of updates and assuming that it was fixed, as expected, by #6334

/close
